sklearn/tree/_splitter.pyx

```diff
@@ -233,18 +234,34 @@ cdef class BaseDenseSplitter(Splitter):
     cdef SIZE_t X_sample_stride
     cdef SIZE_t X_fx_stride
 
+    cdef np.ndarray X_idx_sorted
+    cdef INT32_t* X_idx_sorted_ptr
+    cdef SIZE_t X_idx_sorted_stride
+    cdef SIZE_t n_total_samples
+    cdef SIZE_t* sample_mask
+
     def __cinit__(self, Criterion criterion, SIZE_t max_features,
                   SIZE_t min_samples_leaf, double min_weight_leaf,
-                  object random_state):
+                  object random_state, bint presort):
 
         self.X = NULL
         self.X_sample_stride = 0
         self.X_fx_stride = 0
+        self.X_idx_sorted_ptr = NULL
+        self.X_idx_sorted_stride = 0
+        self.sample_mask = NULL
+        self.presort = presort
```

Comment:
Setting presort should raise an error for splitters that doesn't support presort.